plugins/chat/app/services/chat/create_message_interaction.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Chat
+  # Service responsible for creating and validating a new interaction between a user and a message.
+  #
+  # @example
+  #  Chat::CreateMessageInteraction.call(params: { message_id: 3, action_id: "xxx" }, guardian: guardian)
+  #
+  class CreateMessageInteraction
+    include ::Service::Base
+
+    # @!method self.call(guardian:, params:)
+    #   @param [Guardian] guardian
+    #   @param [Hash] params
+    #   @option params [Integer] :message_id
+    #   @option params [Integer] :action_id
+    #   @return [Service::Base::Context]
+    params do
+      attribute :message_id, :integer
+      validates :message_id, presence: true
+
+      attribute :action_id, :string
+      validates :action_id, presence: true
+    end
+
+    model :message
+    policy :can_interact_with_message
+    model :action
+
+    transaction do
+      model :interaction
+      step :trigger_interaction
+    end
+
+    private
+
+    def fetch_message(params:)
+      Chat::Message.find_by(id: params.message_id)
+    end
+
+    def fetch_action(params:, message:)
+      message.blocks&.each do |item|
+        found_element = item["elements"].find { |element| element["action_id"] == params.action_id }
+        return found_element if found_element
+      end
+      nil
```

Comment:
You can nest `find` ;)

```suggestion
      message.blocks&.find do |item|
        item["elements"].find { |element| element["action_id"] == params.action_id }
      end
```

Reply:
![image](https://github.com/user-attachments/assets/39ac15fe-7148-4196-b5d7-8655cd607c9d)

No it doesn't :p because that means it will return the block, and not the element.